lib/ansible/modules/cloud/vmware/vmware_host_hyperthreading.py

```diff
@@ -0,0 +1,246 @@
+from ansible.module_utils._text import to_native
+
+
+class VmwareHostHyperthreading(PyVmomi):
+    """
+    Class to manage Hyperthreading for an ESXi host system
+    """
+    def __init__(self, module):
+        super(VmwareHostHyperthreading, self).__init__(module)
+        cluster_name = self.params.get('cluster_name')
+        esxi_host_name = self.params.get('esxi_hostname')
+        self.hosts = self.get_all_host_objs(cluster_name=cluster_name, esxi_host_name=esxi_host_name)
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system with given configuration.")
+
+    def ensure(self):
+        """
+        Function to manage Hyperthreading for an ESXi host system
+        """
+        results = dict(changed=False, result=dict())
+        enable_hyperthreading = self.params.get('enabled')
+        host_change_list = []
+        for host in self.hosts:
+            changed = False
+            results['result'][host.name] = dict(msg='', error='')
+
+            hyperthreading_info = host.config.hyperThread
+
+            if enable_hyperthreading:
+                results['result'][host.name]['desired_state'] = "enabled"
+                # Don't do anything if Hyperthreading is already enabled
+                if hyperthreading_info.config and hyperthreading_info.active:
+                    results['result'][host.name]['previous_state'] = "enabled"
+                    results['result'][host.name]['current_state'] = "enabled"
+                    results['result'][host.name]['msg'] = "Hyperthreading is already enabled and active for host '%s'" % \
+                                                          host.name
+                elif hyperthreading_info.config and hyperthreading_info.active is False:
+                    changed = True
+                    results['result'][host.name]['previous_state'] = "enabled"
+                    results['result'][host.name]['current_state'] = "enabled"
+                    results['result'][host.name]['msg'] = ("Hyperthreading is already enabled for host '%s', but not active."
+                                                           " A reboot is required!" % host.name)
+                # Enable Hyperthreading
+                else:
+                    # Check if Hyperthreading is available
+                    if hyperthreading_info.available:
+                        if not self.module.check_mode:
+                            try:
+                                host.configManager.cpuScheduler.EnableHyperThreading()
+                                changed = True
+                                results['result'][host.name]['changed'] = True
+                                results['result'][host.name]['previous_state'] = "disabled"
+                                results['result'][host.name]['current_state'] = "enabled"
+                                results['result'][host.name]['msg'] = "Hyperthreading enabled for host '%s'" % host.name
```

Comment:
It would be good to call out a reboot is required to make it active

Reply:
I did that too. Thanks.